packages/next/next-server/server/config.ts

```diff
@@ -208,6 +211,41 @@ function assignDefaults(userConfig: { [key: string]: any }) {
     }
   }
 
+  if (result?.experimental?.images) {
+    const { images } = result.experimental
+    if (typeof images !== 'object') {
+      throw new Error(
+        `Specified images should be an object received ${typeof images}`
+      )
+    }
+    if (images.domains) {
+      if (!Array.isArray(images.domains)) {
+        throw new Error(
+          `Specified images.domains should be an Array received ${typeof images.domains}`
+        )
+      }
+      const invalid = images.domains.find((d: unknown) => typeof d !== 'string')
+      if (invalid) {
+        throw new Error(
+          `Specified images.domains should be an Array of strings received Array of ${typeof invalid}s`
+        )
+      }
+    }
```

Comment:
```suggestion
      const invalid = images.domains.filter((d: unknown) => typeof d !== 'string')
      if (invalid.length > 0) {
        throw new Error(
          `Specified images.domains should be an Array of strings received invalid values (${invalid.join(', ')})`
        )
      }
    }
```
There could be multiple invalid values that could be helpful to list all at once